community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/RotationStrategy.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.store.kvstore;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
+import org.neo4j.helpers.Pair;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.pagecache.PageCache;
+
+abstract class RotationStrategy<Meta>
+{
+    protected final FileSystemAbstraction fs;
+    protected final PageCache pages;
+    private final ProgressiveFormat<Meta> format;
+
+    RotationStrategy( FileSystemAbstraction fs, PageCache pages, ProgressiveFormat<Meta> format )
+    {
+        this.fs = fs;
+        this.pages = pages;
+        this.format = format;
+    }
+
+    protected abstract File initialFile();
+
+    protected abstract Iterable<File> candidateFiles();
+
+    protected abstract File nextFile( File previous );
+
+    public final Pair<File, KeyValueStoreFile<Meta>> open() throws IOException
+    {
+        KeyValueStoreFile<Meta> result = null;
+        File path = null;
+        for ( File candidatePath : candidateFiles() )
+        {
+            KeyValueStoreFile<Meta> file;
+            try
+            {
+                file = format.openStore( fs, pages, candidatePath );
+            }
+            catch ( Exception e )
```

Comment:
No, because semantically it doesn't make sense to throw an `IOException` when the format turns out to be invalid, or the checksum doesn't match up.